src/main/java/com/trolley/trolley/InvoiceGateway.java

```diff
@@ -141,23 +141,23 @@ public List<Invoice> search(final SearchBy searchBy,
             case INVOICE_ID:
             case RECIPIENT_ID:
             case INVOICE_NUMBER:
+            case EXTERNAL_ID:
             case TAGS:
                 if(null == paramsList){
                     throw new InvalidFieldException("variable paramsList can not be null for the provided searchBy parameter. Refer to method's Javadoc for more details.");
                 }
-                body = "{\""+searchBy.name()+"\":"
+                body = "{\""+searchBy.getKey()+"\":"
                     +new ObjectMapper()
                         .setSerializationInclusion(Include.NON_EMPTY)
                         .writeValueAsString(paramsList)
                 +"}";
```

Comment:
Instead of building it by hand, maybe you would like to do it somehow like this:
``` Java
ObjectMapper objectMapper = new ObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
body = objectMapper.writeValueAsString(Map.of(searchBy.getKey(), paramsList));
```